### ISSUE TITLE
Проблемы с назначением клавиши CAPS

### ISSUE THREAD

**Issue by fangelion** [BladeMight/Mahou#143]
Выбрал переключение английской раскладки на CAPS, а русской на CTRL+CAPS.
Срабатывание на CAPS проходит нормально, но функция самого капса не отключается.
Срабатывание на CTRL+CAPS проходит, но тут же раскладка возвращается обратно в английскую. И при этом срабатывает стандартная функция клавиши CAPS. Мне кажется, что при выборе этой клавиши в сочетании для переключения раскладок, или других хоткеев, основная функция должна отключиться.
Windows10 Pro
[2017.11.12.txt](https://github.com/BladeMight/Mahou/files/1465065/2017.11.12.txt)

**Comment by BladeMight** [comment]
@fangelion Я уже думал над этим, и кое-кто уже сообщал подобное #118, там же я объяснил почему пока что так, думал стоит *возродить* low-level hook'и и сделать их более стабильными.

**Comment by BladeMight** [comment]
@fangelion Готово, [Mahou-v2.5.5.0](https://github.com/BladeMight/Mahou/releases/tag/v2.5.5.0), добавил возможность *отключить* CapsLock переназначив ее на F18(галочка во вкладке `Функции`), также не забудьте выключить функцию `таймер CapsLock` которая будет его выключать при Shift/Alt/Ctrl+CapsLock(только если назначена такая гор. клавиша, то CapsLock будет переназначен на F18).

**Comment by rg-software** [comment]
С caps есть ещё одна проблема (видимо, без low level не обойтись...)
Если случайно нажать caps в то время, когда ввод текста недоступен, то Mahou не обрабатывает нажатие, и в итоге получаем caps lock. Соответственно, приходится переключаться куда-нибудь, где Mahou не работает, и снова нажимать caps lock, чтобы его отключить. Очень неприятно.

**Comment by BladeMight** [comment]
> видимо, без low level не обойтись

Он уже есть в коде....

> Если случайно нажать caps в то время, когда ввод текста недоступен, то Mahou не обрабатывает нажатие

Странно, Mahou - не важно есть ли ввод текста нету, он просто ловит нажатия.

> где Mahou не работает

Типа админских приложений?

> Соответственно, приходится переключаться куда-нибудь

> и снова нажимать caps lock, чтобы его отключить.


Можно же нажимать <kbd>Shift</kbd>/<kbd>Alt</kbd>/<kbd>Ctrl</kbd>+<kbd>Caps Lock</kbd> для отключения.
P.S. В случае с IME(например Google Japanese IME, у меня просто у самого стоит) не сработает т.к. они зарезервированы для действий IME, и отключение на будет работать при этой(IME) раскладке.

**Comment by rg-software** [comment]
> Типа админских приложений?
Да... ну да с этим жить можно, не так часто приходится их использовать. К сожалению, аналогичный эффект случается и в других случаях, я попробую воспроизвести, Мне казалось, что дело в отсутствии поля ввода, но вижу, что был неправ. Видимо, причина в другом (может, процессор перегружен и не отреагировал).

>Можно же нажимать Shift/Alt/Ctrl+Caps Lock для отключения.
Да, действительно.
